main.tf

```diff
@@ -14,19 +14,19 @@ resource "aws_efs_file_system" "this" {
   throughput_mode                 = var.throughput_mode
 
   dynamic "lifecycle_policy" {
-    for_each = [for k, v in var.lifecycle_policy : { (k) = v }]
+    for_each = { for k, v in var.lifecycle_policy : k => v if v != null }
 
     content {
-      transition_to_ia                    = try(lifecycle_policy.value.transition_to_ia, null)
-      transition_to_archive               = try(lifecycle_policy.value.transition_to_archive, null)
-      transition_to_primary_storage_class = try(lifecycle_policy.value.transition_to_primary_storage_class, null)
+      transition_to_ia                    = lifecycle_policy.key == "transition_to_ia" ? lifecycle_policy.value : null
```

Comment:
with the variable optional attributes these all simply become

```hcl
transition_to_ia                    = lifecycle_policy.value.transition_to_ia
transition_to_archive               = lifecycle_policy.value.transition_to_archive
transition_to_primary_storage_class = lifecycle_policy.value.transition_to_primary_storage_class
```

Reply:
Same for here, this kept the original logic:
```
      + lifecycle_policy {
          + transition_to_ia = "AFTER_30_DAYS"
        }
      + lifecycle_policy {
          + transition_to_primary_storage_class = "AFTER_1_ACCESS"
        }
```